README.md
## Welcome to the Bronze age repository.

changelog.txt
Build: 1.15.2-31.0.9 - Wed Jan 29 21:11:24 GMT 2020
	40495288+Yamahari:
		Defer material getting for ChestTileEntityRenderer into seperate function (#6465)

=========
Build: 1.15.2-31.0.8 - Wed Jan 29 21:04:44 GMT 2020
	37298044+Cadiboo:
		Small Forge config fixes (#6466)

=========
Build: 1.15.2-31.0.7 - Wed Jan 29 20:50:44 GMT 2020
	lex:
		Fix PlayerChangedDimensionEvent using incorrect from dimension argument in ServerPlayerEntity.teleport. Closes #6457

	lex:
		Fix incorrect keybinding modifiers patch. Closes #6478

	lex:
		Fix ChunkDataEvent.Load not firing, it now fires in the aync thread. Be careful. Closes #6471

=========
Build: 1.15.2-31.0.4 - Wed Jan 29 19:31:50 GMT 2020
	lex:
		Fix WakeupEvent patch misalign. Closes #6474

	lex:
		Fix default mod resources loosing it's sort order.

=========
Build: 1.15.2-31.0.2 - Wed Jan 29 17:26:58 GMT 2020
	goto.minecraft:
		[1.15] Reimplemented the ITeleporter interface (#6404)

=========
Build: 1.15.2-31.0.1 - Thu Jan 23 21:59:18 GMT 2020
	David Quintana:
		Hook "gui_light" value from the model into the IBakedModel implementations.
		Apply some of the suggestions from the 1.14 models PR.

=========
Build: 1.15.2-31.0.0 - Wed Jan 22 21:47:51 GMT 2020
	lex:
		1.15.2 Initial Update

=========
Build: 1.15.1-30.0.51 - Wed Jan 22 21:10:07 GMT 2020
	ichttt:
		Cleanup fml packages removing old deprecated classes related to GUIs/Configs and fix HoverChecker (#6437)
		
		Fixed ModListWidget name to avoid confusion and collision.
		Delete CheckBox, as vanilla has it's own CheckBox now which looks much better
		Rename ModConfigEvent.ConfigReloading to ModConfigEvent.Reloading

=========
Build: 1.15.1-30.0.50 - Wed Jan 22 20:38:22 GMT 2020
	12281102+Aeltumn:
		Fixed parameter name issue in ISTER rename. #6461 (#6468)

=========
Build: 1.15.1-30.0.49 - Wed Jan 22 19:56:16 GMT 2020
	lex:
		Fix default widget foreground color.

=========
Build: 1.15.1-30.0.48 - Wed Jan 22 19:30:35 GMT 2020
	diesieben07:
		Use extended version of getLightValue in all places (#6433)

	Bward7864:
		Fixed RenderTypeLookup loosing mappings for registry replacements. (#6452)

	wynprice999:
		Fixed rounding issue with entity navigation fix (#6455) (#6456)

	37298044+Cadiboo:
		Fix Widget Foreground Color not allowing pure black (#6460)

	12281102+Aeltumn:
		Updated method names referencing ItemStackTileEntityRenderer (#6461)

=========
Build: 1.15.1-30.0.43 - Wed Jan 22 19:22:15 GMT 2020
	scottehboeh:
		[1.15.x] Added entity nameplate rendering event hook (#6416)

=========
Build: 1.15.1-30.0.42 - Tue Jan 21 03:47:02 GMT 2020
	tterrag:
		Remove emissive items hook, add small utility for bakedquad lighting

=========
Build: 1.15.1-30.0.41 - Thu Jan 16 02:30:04 GMT 2020
	David Quintana:
		Fix breaking overlay weirdness. applyBakedNormals was not transforming the baked normals when applying them. Fixes #6432

=========
Build: 1.15.1-30.0.40 - Thu Jan 16 00:30:10 GMT 2020
	12281102+Aeltumn:
		[1.15] Add new InputEvent.ClickInputEvent, addresses #5802 (#6047)

=========
Build: 1.15.1-30.0.39 - Mon Jan 13 22:21:01 GMT 2020
	David Quintana:
		Fix dark spots in fullbright quad rendering.
		Fix multi-layer model not declaring dependency on the particle texture.
		Fix mistake in 2c054fca which I didn't intend to push when I did.

=========
Build: 1.15.1-30.0.38 - Mon Jan 13 18:39:06 GMT 2020
	David Quintana:
		Attempt to fix fullbright lighting.
		
		Fix lightmap value calculation.

	David Quintana:
		Fix multi-layer model loader to actually work in a sane way. No more "base" model involved, and now transforms and overrides work as expected.

=========
Build: 1.15.1-30.0.36 - Thu Jan 09 22:28:02 GMT 2020
	tterrag:
		Add constructor for FishBucketItem that accepts suppliers

=========
Build: 1.15.1-30.0.35 - Tue Jan 07 22:58:54 GMT 2020
	David Quintana:
		Fix QuadTransformer normal processing.

	Choonster.2010:
		Fix validation in RotationBuilder#angle (fixes #6323) (#6408)

=========
Build: 1.15.1-30.0.33 - Tue Jan 07 22:20:46 GMT 2020
	richard:
		Add support for custom nether portal frame blocks (#6389)

=========
Build: 1.15.1-30.0.32 - Tue Jan 07 21:22:49 GMT 2020
	RANKSHANK:
		InputEvent.RawMouseEvent fix (#6395)

=========
Build: 1.15.1-30.0.31 - Tue Jan 07 21:15:19 GMT 2020
	DaemonUmbra:
		Fixed Big Mushrooms not generating (#6390)
		
		Removed vanilla negation from Forge logic that negated the vanilla logic it was replacing
		also ran datagen to update the dirt tag to reflect changes that were previously made in the datagen code

=========
Build: 1.15.1-30.0.30 - Mon Jan 06 00:04:17 GMT 2020
	tterrag:
		Fix dyes tag containing dye items instead of tags
		
		Update generated data

=========
Build: 1.15.1-30.0.29 - Sun Jan 05 19:36:08 GMT 2020
	Choonster.2010:
		[1.15.x] Remove unused parameter from ElementBuilder#rotation (fixes #6321) (#6407)

=========
Build: 1.15.1-30.0.28 - Sun Jan 05 19:29:10 GMT 2020
	tterrag:
		Refactor BlockStateProvider to encapsulate a BlockModelProvider
		
		Make most protected methods of data generators public

=========
Build: 1.15.1-30.0.27 - Sun Jan 05 18:39:00 GMT 2020
	tterrag:
		Fix crash when building BakedQuadBuilder
		
		Add javadocs and move sprite to constructor where possible

=========
Build: 1.15.1-30.0.26 - Sun Jan 05 05:05:58 GMT 2020
	tterrag:
		Fix incorrect check for missing texture in DynamicBucketModel

=========
Build: 1.15.1-30.0.25 - Sun Jan 05 04:38:48 GMT 2020
	tterrag:
		Remove custom vertex format from BakedQuad

=========
Build: 1.15.1-30.0.24 - Sat Jan 04 16:19:24 GMT 2020
	David Quintana:
		Cherry-pick ca819e09 from the 1.14.x branch. Removes a patch from NativeImage which was invalid.

	David Quintana:
		A few more binary-breaking changes, before it's too late:
		- Move CompositeModel out of its package.
		- Move a few inner classes from the OBJ loader.
		- Rename the bucket model and fluid model classes to match the naming convention.

src/main/java/com/yanny/age/bronze/blocks/BoostedFurnaceGui.java
package com.yanny.age.bronze.blocks;

import com.yanny.age.bronze.Reference;
import net.minecraft.client.gui.screen.inventory.ContainerScreen;
import net.minecraft.entity.player.PlayerInventory;
import net.minecraft.util.ResourceLocation;
import net.minecraft.util.text.ITextComponent;

import static com.yanny.age.bronze.blocks.BoostedFurnaceTileEntity.*;

public class BoostedFurnaceGui extends ContainerScreen<BoostedFurnaceContainer> {
    private static final ResourceLocation GUI = new ResourceLocation(Reference.MODID, "textures/gui/container/boosted_furnace.png");

    private BoostedFurnaceContainer container;

    public BoostedFurnaceGui(BoostedFurnaceContainer screenContainer, PlayerInventory inv, ITextComponent titleIn) {
        super(screenContainer, inv, titleIn);
        this.container = screenContainer;
    }

    @Override
    public void render(int mouseX, int mouseY, float partialTicks) {
        this.renderBackground();
        super.render(mouseX, mouseY, partialTicks);
        this.renderHoveredToolTip(mouseX, mouseY);
    }

    @Override
    protected void drawGuiContainerBackgroundLayer(float partialTicks, int mouseX, int mouseY) {
        if (minecraft == null) {
            return;
        }

        minecraft.getTextureManager().bindTexture(GUI);
        blit(guiLeft, guiTop, 0, 0, xSize, ySize);

        if (container.getBurnTimeLeft() > 0) {
            int k = Math.round((container.getBurnTimeLeft() / (float) container.getBurnTimeTotal()) * 14);
            this.blit(guiLeft + 26, guiTop + 36 + 14 - k, 176, 14 - k, 14, k + 1);
        }

        if (container.getTemperature() > 0) {
            int k = Math.round((container.getTemperature() / (float) (MAX_NO_CHIMNEY_TEMPERATURE + MAX_TEMPERATURE_PER_CHIMNEY * MAX_CHIMNEY_COUNT)) * 56);
            this.blit(guiLeft + 59, guiTop + 15 + 56 - k, 190, 56 - k, 21, k + 1);
        }

        if (container.getProcessLeft() <= MELTING_TICKS) {
            int k = 24 - Math.round((container.getProcessLeft() / (float) MELTING_TICKS) * 24);
            this.blit(guiLeft + 93, guiTop + 35, 211, 0, k + 1, 17);
        }

        drawString(font, container.getTemperature() + " °C", guiLeft + 86,guiTop + 56, -1);

        if (container.getMeltingPoint() < 2000) {
            drawString(font, "> " + container.getMeltingPoint() + " °C", guiLeft + 82, guiTop + 18, -1);
        }
    }
}
